src/features/hover.rs

```diff
@@ -0,0 +1,390 @@
+                None,
+                None,
+            ))
+        }
+        _ => None,
+    }
+}
+
+/// Formats the CSS entry into a hover content string.
+///
+/// # Arguments
+///
+/// * `name` - The name of the CSS entity.
+/// * `desc` - An optional description of the entity.
+/// * `syntax` - An optional syntax string.
+/// * `specificity` - An optional specificity string.
+/// * `browsers` - An optional browsers string.
+/// * `reference` - An optional reference URL.
+/// * `restriction` - An optional restriction string.
+///
+/// # Returns
+///
+/// A `String` containing the formatted hover content in Markdown.
+fn format_css_entry(
+    name: &str,
+    desc: Option<&str>,
+    syntax: Option<&str>,
+    specificity: Option<(u32, u32, u32)>,
+    browsers: Option<&str>,
+    reference: Option<&str>,
+    restriction: Option<&str>,
+) -> String {
+    let mut content = String::new();
+    content.push_str(&format!("**{}**\n\n", escape_markdown(name)));
+
+    // Add the description if available
+    if let Some(description) = desc {
+        content.push_str(description);
+        content.push_str("\n\n");
+    }
+
+    // Add syntax if available
+    if let Some(syntax) = syntax {
+        content.push_str(&format!("**Syntax**: `{}`\n\n", syntax));
+    }
+
+    // Add specificity if available
+    if let Some((ids, classes, elements)) = specificity {
+        content.push_str(&format!(
+            "[Selector Specificity](https://developer.mozilla.org/docs/Web/CSS/Specificity): ({}, {}, {})\n\n",
+            ids, classes, elements
+        ));
+    }
+
+    // Add restriction if available
+    if let Some(info) = restriction {
+        content.push_str(&format!("**Restriction**: {}\n\n", info));
+    }
+
+    // Add browsers if available
+    if let Some(browsers) = browsers {
+        content.push_str("**Supported Browsers**:\n");
+        for browser in browsers.split(',') {
+            content.push_str(&format!("- {}\n", browser.trim()));
+        }
+        content.push('\n');
+    }
+
+    // Add reference if available
+    if let Some(reference) = reference {
+        content.push_str("**Reference**:\n");
+        content.push_str(&format!("- [{}]({})\n\n", name, reference));
+    }
+
+    content
+}
+
+fn escape_markdown(text: &str) -> String {
+    text.replace('*', "\\*")
+}
+
+/// Given a CSS selector, calculates the specificity of the selector.
+/// The specificity is returned as a tuple of (ids, classes, elements).
+/// Refer to the MDN documentation for more information:
+/// https://developer.mozilla.org/docs/Web/CSS/Specificity
+///
+/// # Arguments
+/// - `selector` - The CSS selector to calculate the specificity for.
+///
+/// # Returns
+/// A tuple containing the specificity of the selector as (ids, classes, elements).
+fn calculate_specificity(selector: &str) -> (u32, u32, u32) {
+    let mut ids = 0;
+    let mut classes = 0;
+    let mut elements = 0;
```

Comment:
```suggestion
	let (mut ids, mut classes, mut elements) = (0, 0, 0);
```

You can destructure like this instead of doing three lines, this is a bit more cohesive with the fact that the function itself returns a tuple (and I believe more idiomatic as well)